vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -35,6 +36,17 @@ class Action(VizroBaseModel):
         regex="^[a-zA-Z0-9_]+[.][a-zA-Z_]+$",
     )
 
+    # TODO: Abstract away once pattern becomes clear regarding function/import checks
+    @validator("function")
+    def check_export_imports(cls, function):
+        if (
+            function._function.__name__ == "export_data"
+            and function._arguments.get("file_format") == "xlsx"
+        ):
+            if importlib.util.find_spec("openpyxl") is None and importlib.util.find_spec("xlsxwriter") is None:
```

Comment:
We had the discussion about predefined actions error handling few months ago (and we are aware we still don't have the right answer and solution). I don't have a strong preference about where these errors should be handled. There are two options:
 - In the Action model (error will be raised as soon the Action model is initialised).
 - In the predefined action function body (error will be dynamically raised when the action is triggered from the UI). 

There's just one thing I have a strong preference and it is that we need to be consistent.

There are already few exceptions handling defined in the predefined action functions and the best example is the one from the predefined `export_data` action where we handle "Unknown file_format":

```
        if file_format == "csv":
            writer = data_frames[target_id].to_csv
        elif file_format == "xlsx":
            writer = data_frames[target_id].to_excel
        else:
            raise ValueError(f'Unknown "file_format": {file_format}.' f' Known file formats: "csv", "xlsx".')
```
 
This exception could also be handled in a similar way the required installed library is (in the Action model validator).

Do you guys think this error is more specific than "Unknown file_format" is and that it should be handled here (in a different way)?

Reply:
Hm that's a very valid point. I just tried it out, and it is a very similar case to the one we just implemented. My gut feeling is also that we should be consistent, and having the wrong parameters feels like a similar level to not having a module installed...

@antonymilne: What do you think? Should we address this immediately, or shall we merge as is to come back to this later?

Reply:
@petar-qb We have moved this both to the Action now (including a TODO on how to improve that).

Could we cover this validator in the Action model unit tests?

Reply:
That's great 🥇.  

The PR is ready for merging (just remove or comment out the `tests/unit/vizro/actions/test_export_data_action.py::TestExportData::test_invalid_file_format`, and I'll incorporate this test in my PR).

I also think we need to talk more about action function arguments validation (really not urgent), because the topic is slightly complicated then it seems on the first glance.